### DISCUSSION TITLE
Stars rating question support

### DISCUSSION THREAD

**Discussion by rgmerck** [surveydown-dev/surveydown#109] [Feature Requests]
### Description

SImple feature request, a lot of surveys typically include the possibility of giving a very simple feedback in the form of stars, where the user is able to click on the right number of stars to give a "low" to "positive" rating for something. It would be great if you could implement something like this.

**Comment by jhelvy** [comment]
I like this! @pingfan-hu , any thoughts? I'm guessing this could be relatively easy to support by modifying a regular `mc` type question.

**Reply by pingfan-hu** [reply]
Yeah I think it's a great idea. We may need a UI similar to [sweetalert](https://sweetalert.js.org), and then automatically collect users' ratings in our data sheet.

**Comment by pingfan-hu** [comment]
Hi @rgmerck ! The new feature is done. You can try it out [here](https://surveydown.shinyapps.io/types/).
In this demo survey, you can simply click on "Skip to the end" on the 2nd page. It will deliver you to the end. Then please click on the "Exit Survey" button (or simply push your "Enter" key) so that you'll see the rating question.
You can also continuously push your "Enter" key while you land on the survey, since it triggers the "Next" button and delivers you to the next pages.

If you want to implement it, you can define in `sd_server()`:
```r
sd_server(
  rate_survey = TRUE
)
```
The value of `rate_survey` is default to `FALSE` so you need to define it as `TRUE`.

<img width="559" alt="image" src="https://github.com/user-attachments/assets/7008b6db-36a1-4e59-9fc5-50ebdb62e87d">

**Comment by pingfan-hu** [comment]
[Rating question docu](https://surveydown.org/config-options#rating-question) is updated.

**Comment by jhelvy** [comment]
This is now supported as of v0.3.4 using the `rate_survey` argument in `sd_server()`.
